BrainPortal/cbrain_plugins/cbrain-plugins-base/userfiles/single_file/single_file.rb

```diff
@@ -80,4 +80,25 @@ def gzip_content(operation = :compress) #:nodoc:
     self.sync_to_provider
   end
 
+  # compute file md5 hash sum (useful to avoid tampering, e.g. with task archive)
+  def cache_compute_md5sum
+    md5command =
+        case CBRAIN::System_Uname
+        when /Linux/i
+          "md5sum"
+        when /Darwin/i
+          "md5"
+        else
+          "md5sum"
+        end
+    self.sync_to_cache
+
+    content_path = self&.cache_full_path
+    cb_error("Can't find cache path for file #{self.name}") if content_path.blank?
```

Comment:
This cb_error is unncessary, the cache_full_path is guaranteed to be returned.

It also means that the `&.` at the previous line is also unnecessary

Reply:
I thought it depend on dp implementation, but removed